Tests/GitHubContents.tests.ps1

```diff
@@ -335,6 +335,67 @@ try
             }
         }
 
+        Context 'When setting new file content by path' {
+            BeforeAll {
+                $filePath = 'notes'
+                $fileName = 'hello.txt'
+                $commitMessage = 'Commit Message'
+                $content = 'This is the content for test.txt'
+                $branchName = 'master'
+                $committerName = 'John Doe'
+                $committerEmail = 'john.doe@testdomain.com'
+                $authorName = 'Jane Doe'
+                $authorEmail = 'jane.doe@testdomain.com'
+
+                $contentPath = New-TemporaryFile
+                $content | Out-File -FilePath $contentPath
+
+                $setGitHubContentParms = @{
+                    Path = "$filePath/$fileName"
+                    CommitMessage = $commitMessage
+                    Branch = $branchName
+                    ContentPath = $contentPath
+                    Uri = $repo.svn_url
+                    CommitterName = $committerName
+                    CommitterEmail = $committerEmail
+                    authorName = $authorName
+                    authorEmail = $authorEmail
+                }
+
+                $result = Set-GitHubContent @setGitHubContentParms -PassThru
+            }
+
+            It 'Should have the expected type and additional properties' {
+                $result.PSObject.TypeNames[0] | Should -Be 'GitHub.Content'
+                $result.content.name | Should -Be $fileName
+                $result.content.path | Should -Be "$filePath/$fileName"
+                $result.content.url | Should -Be ("https://api.github.com/repos/$($script:ownerName)" +
+                    "/$repoName/contents/$filePath/$($fileName)?ref=$BranchName")
+                $result.commit.author.name | Should -Be $authorName
+                $result.commit.author.email | Should -Be $authorEmail
+                $result.commit.committer.name | Should -Be $committerName
+                $result.commit.committer.email | Should -Be $committerEmail
+                $result.commit.message | Should -Be $commitMessage
+            }
+
+            It 'Should have written the correct content' {
+                $getGitHubContentParms = @{
+                    Path = "$filePath/$fileName"
+                    Uri = $repo.svn_url
+                    MediaType = 'Raw'
+                    ResultAsString = $true
+                }
+
+                $writtenContent = Get-GitHubContent @getGitHubContentParms
```

Comment:
Hmmmm....I _think_ that running another command while within the `It` block goes against the Pester 5 guidelines.   @tigerfansga, can you chime in here on this?  I want to make sure that we're not doing something that is going to break all your hard work in converting the tests over.

Reply:
> Hmmmm....I _think_ that running another command while within the `It` block goes against the Pester 5 guidelines. @tigerfansga, can you chime in here on this? I want to make sure that we're not doing something that is going to break all your hard work in converting the tests over.

@HowardWolosky this should be fine. Running commands in an It block is fine. The item to be careful is Pester v5 creates a new scope before calling an It block. So any variables defined in the It block are limited to that It block. In this case, he is only using the variable inside a single It block. See the *Run* section of [Discovery and Run](https://pester.dev/docs/usage/discovery-and-run)

This code could be consolidated to 
```powershell
$content | Should -Be (Get-GitHubContent @getGitHubContentParms)
```
If any test changes are merged before my changes are merged, I will likely need to update my branch and validate the changes work as merged.

Reply:
Let me know if anything needs to change here.